src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -148,11 +152,21 @@ public ResultResource(ProgrammingExerciseParticipationService programmingExercis
         try {
             planKey = continuousIntegrationService.get().getPlanKey(requestBody);
         }
-        // TODO: How can we catch a more specific exception here? Because of the adapter pattern this is always just Exception...
+        catch (BitbucketException bitbucketException) {
+            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {} :"
+                    + "BitBucket encountered an Exception while trying to retrieve the build plan ", bitbucketException, requestBody);
+            return badRequest();
+        }
+        catch (JenkinsException jenkinsException) {
+            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {}:"
+                    + " The full name of the Jenkins build plan was malformed!", jenkinsException, requestBody);
+            return badRequest();
+        }
```

Comment:
I would suggest merging both catch blocks into one block, which catches `ContinuousIntegrationException`. `ResultResource` should not have to be aware of what kind of CI service is used. Furthermore `BambooService` and `JenkinsService` now wrap all the exceptions into a `ContinuousIntegrationException`, so you can adjust the signature of the `getPlanKey` method in the `ContinuousIntegrationService` interface accordingly. Then you can have a simple try catch block in the `ResultResource`:
```java
try {
    planKey = continuousIntegrationService.get().getPlanKey(requestBody);
} catch (ContinuousIntegrationException e) {
    // log the error and return bad request here
}
```